### PR TITLE
How to represent multiple APIs using one resource server and the implicit flow

### PR DESCRIPTION
**Pull request by khorne3**

### CODE REVIEW

**Comment by khorne3** [comment]
@mpaktiti The sample I used to document this uses the implicit flow, but it looks like it can be used w/ any of our grants. As such, I included links to various options for users at the bottom--what do you think?

**Review by mpaktiti — CHANGES_REQUESTED** [review]
This is good but you don't use the sample at all. I would expect: a short description at the top explaining the scenario (in this scenario we will implement 2 APIs: contacts and calendars, etc), some info on how to run the sample and maybe a screenshot from the SPA, link to download/fork the sample (which should be copied to `auth0-samples` since `auth0-dse` is a private repo https://github.com/auth0/auth0-dse/tree/master/samples/api-auth-implicit

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+  description: How to use multiple APIs and represent them as a single Resource Server in Auth0.
+---
+
+# Represent Multiple APIs as a Single Resource Server
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: The term `Resource Server` is a bit confusing to some. It's OAuth 2.0 terminology, most people just look for "API". I would change the title to sth like `How to define multiple APIs in Auth0`

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+If you have multiple APIs, you can treat them as a single Resource Server. Doing this allows you to implement just one authentication flow. You can then control access to the individual APIs by assigning the appropriate scopes.
+
+This article shows you how to use and represent multiple APIs as a single Resource Server in Auth0 using a [sample application you can download](https://github.com/auth0-samples/auth0-api-auth-implicit-sample) if you would like to follow along as you read.
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: I'm thinking that we could also add the box the Quickstarts use to download/fork project. Wdyt?

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+---
+  description: How to use multiple APIs and represent them as a single Resource Server in Auth0.
+---
+
+# Represent Multiple APIs as a Single Resource Server
+
+If you have multiple APIs, you can treat them as a single Resource Server. Doing this allows you to implement just one authentication flow. You can then control access to the individual APIs by assigning the appropriate scopes.
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: I would rephrase this a bit, to avoid using the "Resource Server" term. Sth like `In most cases, you can [create a single API in the dashboard](/apis) to represent all of your existing APIs. Doing this allows you to implement just one authentication flow. You can then control access to the individual APIs by assigning the appropriate scopes.`

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+This article shows you how to use and represent multiple APIs as a single Resource Server in Auth0 using a [sample application you can download](https://github.com/auth0-samples/auth0-api-auth-implicit-sample) if you would like to follow along as you read.
+
+## The Sample Application
+
+In the sample application, we have:
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: In the sample application you only have the first bullet, the two API js files. The other entries are things involved in this sample application (terms and concepts). 

I would remove this and add a couple of sentences describing the sample: `The sample contains 1 SPA and 2 APIs: contacts and calendar. We will represent the APIs in Auth0 with a single API, which we will name Organizer Service. Then we will configure the scopes .... and see how you can use the Implicit Grant to access these two APIs from the SPA.`

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+For our sample application, we'll add two scopes:
+
+* `read:calendar`;
+* `read:contacts`.
+
+You can think of each one as a microservice.
+
+![](/media/articles/api-auth/tutorials/multiple-apis-one-resource-server/new-scopes.png)
+
+At this point, you've:
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: I would remove this, it looks redundant. Just add something like "add these two scopes and save your changes"

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+You are now ready to provide access to your APIs by granting access tokens to the Auth0 API. By including specific scopes, you can control a client's application to some or all of the APIs represented by the Auth0 API.
+
+:::panel-info Authorization Flows
+
+The rest of this article covers use of the [Implicit Grant](/api-auth/grant/implicit) to reflect the sample. You can, however, use whichever flow best suits your needs.
+
+* If you have a **Non Interactive Client**, you can authorize it to request access tokens to your API by executing a [client credentials exchange](/api-auth/grant/client-credentials).
+* If you are building a **Native App**, you can implement the use of [Authorization Codes using PKCE](/api-auth/grant/authorization-code-pkce).
+:::
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: I would add in the panel `For a full list refer to [API Authorization](/api-auth)`

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+The app initiates the flow and redirects the browser to Auth0 (specifically to the ``/authorize` endpoint), so the user can authenticate.
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: One extra char here: specifically to the ``/authorize` endpoint)

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+```text
+https://YOUR_AUTH0_DOMAIN/authorize?
+scope=read:contacts%20read:calendar&
+audience=organize&
+response_type=id_token%20token&
+client_id=YOUR_CLIENT_ID&
+redirect_uri=http://localhost:3000&
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: We never created a client in Auth0 for the SPA. You need to add this step and set the callback URL to `http://localhost:3000`

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+nonce=NONCE
+```
+The SPA executes this call whenever the user clicks **Login**.
+
+![SPA Home before Login](/media/articles/api-auth/tutorials/multiple-apis-one-resource-server/home.png)
+
+:::panel-info Request Parameters
+For additional information on the call's parameters, refer to the [docs on executing an implementing the Implicit Grant](/api-auth/grant/implicit#1-get-the-user-s-authorization).
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: The link does not work, you want the doc under tutorials.

Comment (mpaktiti) [reply articles/api-auth/tutorials/use-one-rs-for-apis.md]: I would remove this however from the panel and add it as a simple sentence under the `/authorize` script.

File: articles/api-auth/tutorials/use-one-rs-for-apis.md (hunk 4 above)
Comment (mpaktiti) [reply articles/api-auth/tutorials/use-one-rs-for-apis.md]: Mention also that the sample uses our Lock widget.

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+:::
+
+Lock handles the login process (you can log in with a user that exists in a [Connection](/identityproviders) enabled for your Auth0 Client).
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: We never created the Client and we didn't enable a connection for it.

File: articles/api-auth/tutorials/use-one-rs-for-apis.md (hunk 8 above)
Comment (mpaktiti) [reply articles/api-auth/tutorials/use-one-rs-for-apis.md]: Don't add lots of details/screenshots, brief steps and link to reference docs for details.

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+![SPA Login](/media/articles/api-auth/tutorials/multiple-apis-one-resource-server/lock.png)
+
+Next, Auth0 authenticates the user. If this is the first time the user goes through this flow, they will be asked to consent to the scopes that are given to the Client (such as post messages or list contacts).
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: I would replace `(such as post messages or list contacts)` with the real scopes for this sample. A screenshot with the consent window would look nice as well here.

File: articles/api-auth/tutorials/use-one-rs-for-apis.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+If the user consents, Auth0 continues the authentication process, and upon completion, redirects them back to the app with an `access_token` in the hash fragment of the URI. The app can now extract the tokens from the hash fragment. In a Single Page Application (SPA) this is be done using JavaScript (see the `getAccessToken` function located in `index.html` for the sample app). The app can then use the `access_token` to call the API on behalf of the user.
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/use-one-rs-for-apis.md]: `see the `getAccessToken` function located in `index.html` for the sample app` : add inline the script maybe?

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Hey, I left some more comments on this.

**Review by mpaktiti — APPROVED** [review]
Looks good 👍

**Comment by mpaktiti** [comment]
Hey @khorne3 I'm going to merge this. Before I do so can you please do the following:

- Add a new doc entry for the update feed.

- Add the doc at the api-auth index https://auth0-docs-content-pr-3777.herokuapp.com/docs/api-auth List it as a bullet under `Calling APIs from Client-side Web Apps` since the sample involves a SPA. Add it at the end, after `Protect against replay attacks`.
